### DISCUSSION TITLE
Is it possible to get the shadcn sidebar to act like the sidebar on shadcn?

### DISCUSSION THREAD

**Discussion by Quozzo** [shadcn-ui/ui#6042] [General]
The sidebar on shadcn is between the header and the footer but it doesn't look like there's an option so that it doesn't stretch across the whole screen. Does anyone have any suggestions?

**Comment by georgeeburt** [comment]
It's quite hard to grasp what you want to achieve with the given context from your post.

But if it's what I think you mean, you can change the `z-index` of the header or footer to bring them above the sidebar.

Something like this:

```css
.header {
  z-index: 10;
}

.footer {
  z-index: 10;
}
```

**Reply by Quozzo** [reply]
That won't affect the height of the sidebar, so any items low in the list would be clipped behind the footer and inaccessible.

**Reply by georgeeburt** [reply]
You can add a bottom margin within the footer.

**Reply by Quozzo** [reply]
The footer isn't a fixed height and varies depending on the screen size to contain all of the links. It might be possible for the header though.

**Reply by georgeeburt** [reply]
Can you not just use the margins in % units? Then you know you'll have no conflicts there with the footer.

**Comment by MeatBoyed** [comment]
I had a similar question a while ago, instead of figuring out how to do it manually I used v0, which gave me mediocre results, and later found [this dashboard template](https://github.com/satnaing/shadcn-admin) that did what I needed.
I would also recommend this free library of [Shadcn components/templates ](https://github.com/birobirobiro/awesome-shadcn-ui)on Github.
Hope this helps you find what you need

**Comment by lassounski** [comment]
I was having the same issue over here.

The behavior is a feature since the side bar is fixed while you scroll.

If you're are okay with the sidebar rolling with the rest of the page you can try this:

Find the keyword **fixed** in src/components/ui/sidebar.tsx and replace it for **absolute**.

**Reply by Andrew-Chen-Wang** [reply]
this doesn't work if your content needs to scroll, and also it is better override the prop in your component rather than the component source code

**Comment by 0xSigmoid** [comment]
If anyone solves this please let me know, an absolute sidebar is a serious downgrade

**Comment by MicKoreaYoutube** [comment]
i think i fixed it with a little bit of effort
1. change the sidebar's position to absoulte
2. remove the h-svh class above
code will be like
```html
<div
  className={cn(
    "duration-200 relative w-[--sidebar-width] bg-transparent transition-[width] ease-linear",
    "group-data-[collapsible=offcanvas]:w-0",
    "group-data-[side=right]:rotate-180",
    variant === "floating" || variant === "inset"
      ? "group-data-[collapsible=icon]:w-[calc(var(--sidebar-width-icon)_+_theme(spacing.4))]"
      : "group-data-[collapsible=icon]:w-[--sidebar-width-icon]"
  )}
/>

<div
  className={cn(
    "duration-200 absolute z-10 hidden h-svh w-[--sidebar-width] transition-[left,right,width] ease-linear md:flex",
    side === "left"
      ? "left-0 group-data-[collapsible=offcanvas]:left-[calc(var(--sidebar-width)*-1)]"
      : "right-0 group-data-[collapsible=offcanvas]:right-[calc(var(--sidebar-width)*-1)]",
    // Adjust the padding for floating and inset variants
    variant === "floating" || variant === "inset"
      ? "p-2 group-data-[collapsible=icon]:w-[calc(var(--sidebar-width-icon)_+_theme(spacing.4)_+2px)]"
      : "group-data-[collapsible=icon]:w-[--sidebar-width-icon] group-data-[side=left]:border-r group-data-[side=right]:border-l",
    className
  )}
  {...props}
>
  <div
    data-sidebar="sidebar"
    className="flex h-full w-full flex-col bg-sidebar group-data-[variant=floating]:rounded-lg group-data-[variant=floating]:border group-data-[variant=floating]:border-sidebar-border group-data-[variant=floating]:shadow"
  >
    {children}
  </div>
</div>

```

you can find this code in line 225 if you havent change anything

**Comment by astegmaier** [comment]
I was able to get the sidebar to bump down so it doesn't overlap the top bar of my app by doing this:

1. Tweak the generated `components/ui/sidebar.tsx` file to split out the `<SidebarWrapper>` component from `<SideBarContext>`.
2. Render the `<SidebarWrapper>` component in your app in the position you want the sidebar to be (i.e. not full-screen)
3. Give the `<SidebarWrapper>` component a `className` of `relative h-full min-h-auto` and the `<Sidebar>` underneath it a `className` of `absolute h-full`

See [https://github.com/astegmaier/playground-shadcn-sidebar](https://github.com/astegmaier/playground-shadcn-sidebar) for a working example

**Reply by ShashankChourasia** [reply]
worked for me.

**Reply by myudak** [reply]
its working tysmm!

**Comment by Xyli0** [comment]
You need to change some properties of the sidebar-container

1. Change sidebar positioning  from fixed to absolute
2. Change height property form h-svh to h-full to span the full height of its container
3. Remove inset-y-0

```html
<div 
    data-slot="sidebar-container"
    className={cn(
             'absolute z-10 hidden h-full w-(--sidebar-width) transition-[left,right,width] duration-200 ease-linear md:flex',
              side === 'left'
                        ? 'left-0 group-data-[collapsible=offcanvas]:left-[calc(var(--sidebar-width)*-1)]'
                        : 'right-0 group-data-[collapsible=offcanvas]:right-[calc(var(--sidebar-width)*-1)]',
              // Adjust the padding for floating and inset variants.
              variant === 'floating' || variant === 'inset'
                        ? 'p-2 group-data-[collapsible=icon]:w-[calc(var(--sidebar-width-icon)+(--spacing(4))+2px)]'
                        : 'group-data-[collapsible=icon]:w-(--sidebar-width-icon) group-data-[side=left]:border-r group-data-[side=right]:border-l',
              className,
              )}
              {...props}
          >
```

**Comment by DesignByOnyx** [comment]
For me, I had a full-width header followed by the sidebar component. I wanted to preserve most of the existing styles to allow for easy upgrades in the future. Here's what I did.

<details open>
    <summary>1. Add an <code>offsetTop</code> prop to the <code>SidebarProvider</code></summary>
    <img width="250" alt="image" src="https://github.com/user-attachments/assets/37100a61-3574-4d1c-9701-75557d34d29e" />
</details>
<details open>
    <summary>2. Create  a new CSS Custom Property (variable) on the <code>sidebar-wrapper</code> and remove the <code>min-h-svh</code></summary>
    <img width="1300" alt="image" src="https://github.com/user-attachments/assets/12fb154a-1577-45e0-9ab4-3b3f2399049a" />
</details>

<details open>
    <summary>3. Update the <code>height</code> and <code>top</code> on the <code>sidebar-container</code> to use the new variable</summary>
    <img width="1205" height="154" alt="image" src="https://github.com/user-attachments/assets/f7b75182-fcb9-44ce-9cdc-a05ed372592b" />
</details>

___

### Here's how I use it
```jsx
const App = () => {
   // You can use refs, effects, and ResizeObserver to get the actual header height if needed
   const headerHeight = 55;

   return (
      <div className="h-svh grid grid-rows-[auto_1fr]">
         <header className="...">...</header>
         <main>
            <SidebarProvider offsetTop={headerHeight}>
               <Sidebar>
                  {/* Sidebar content goes here */}
               </Sidebar>
               <SidebarInset className="flex-1 relative">
                  {/* Main content goes here */}
               </SidebarInset>
            </SidebarProvider>
         </main>
      </div>
   )
}
```
